api/cluster/resource/templater.go

```diff
@@ -281,10 +287,20 @@ func createPredictorSpec(modelService *models.Service, config *config.Deployment
 				},
 			},
 		}
-	case models.ModelTypePyFunc:
+	case models.ModelTypePyFunc, models.ModelTypePyFuncV3:
 		envVars := models.MergeEnvVars(modelService.EnvVars, createPyFuncDefaultEnvVars(modelService))
 		if modelService.Protocol == protocol.UpiV1 {
-			envVars = append(envVars, models.EnvVar{Name: envGRPCOptions, Value: config.PyfuncGRPCOptions})
+			envVars = append(envVars, models.EnvVar{Name: envGRPCOptions, Value: t.deploymentConfig.PyfuncGRPCOptions})
+		}
+		if modelService.EnabledModelObservability && modelService.Type == models.ModelTypePyFuncV3 {
+			pyfuncPublisherCfg := t.deploymentConfig.PyFuncPublisher
+			envVars = append(envVars, models.EnvVar{Name: envPublisherEnabled, Value: strconv.FormatBool(modelService.EnabledModelObservability)})
+			envVars = append(envVars, models.EnvVar{Name: envPublisherKafkaTopic, Value: modelService.GetPredictionLogTopicForVersion()})
+			envVars = append(envVars, models.EnvVar{Name: envPublisherKafkaBrokers, Value: pyfuncPublisherCfg.Kafka.Brokers})
+			envVars = append(envVars, models.EnvVar{Name: envPublisherKafkaLinger, Value: fmt.Sprintf("%d", pyfuncPublisherCfg.Kafka.LingerMS)})
+			envVars = append(envVars, models.EnvVar{Name: envPublisherKafkaAck, Value: fmt.Sprintf("%d", pyfuncPublisherCfg.Kafka.Acks)})
+			envVars = append(envVars, models.EnvVar{Name: envPublisherSamplingRatio, Value: fmt.Sprintf("%f", pyfuncPublisherCfg.SamplingRatioRate)})
```

Comment:
We might need to add configuration for sampling strategy at a later time. For now, we can keep this as it is.